HopsanGenerator/src/generators/HopsanExeGenerator.cpp

```diff
@@ -0,0 +1,375 @@
+    QMap<QString, QString> assetsMap;
+    if (!assets.empty()) {
+        printMessage("Exporting model assets");
+    }
+    for (const auto& asset : assets) {
+        QFileInfo assetInfo(asset.c_str());
+        QString absSourcePath = pSystem->findFilePath(asset).c_str();
+        QString targetPath;
+        if (assetInfo.isAbsolute()) {
+            // For absolute windows paths, replace \ with /
+            targetPath = absSourcePath.replace(R"(\)", "/");
+            // For absolute windows paths, replace :/ with /
+            targetPath = absSourcePath.replace(":/", "/");
+            // For Unix paths remove leading /
+            if (targetPath.startsWith("/")) {
+                targetPath.remove(0,1);
+            }
+        } else {
+            // For relative windows paths, replace \ with /
+            targetPath = assetInfo.filePath().replace(R"(\)", "/");
+            // For relative paths, replace leading ../ with a number, to ensure uniqueness
+            int ctr=0;
+            while(targetPath.startsWith("../")) {
+                ++ctr;
+                targetPath.remove(0,3);
+            }
+            if(ctr>0) {
+                targetPath.prepend(QString("%1_").arg(ctr));
+            }
+        }
+        targetPath.prepend(buildPath+"/resources/");
+        copyFile(absSourcePath, targetPath);
+        assetsMap.insert(asset.c_str(), resourceDir.relativeFilePath(targetPath));
+    }
+
+    bool genOK = generateModelFile(pSystem, buildPath, assetsMap);
+    if (!genOK) {
+        printErrorMessage("Failed to generate model file");
+        return false;
+    }
+
+    //------------------------------------------------------------------//
+    // Compiling and linking
+    //------------------------------------------------------------------//
+
+    if(!compileAndLinkExe(buildPath, modelName, x64))
+    {
+        return false;
+    }
+
+    printMessage("Finished.");
+    return true;
+}
+
+bool HopsanExeGenerator::generateModelFile(const ComponentSystem *pSystem, const QString &buildPath, const QMap<QString,QString>& replaceMap) const
+{
+    QString modelName = pSystem->getName().c_str();
+    QFile modelHppFile(buildPath + "/model.hpp");
+    QFile modelFile(buildPath + "/../" + modelName + ".hmf");
+
+    printMessage("Generating "+modelHppFile.fileName());
+
+    if (!modelFile.open(QIODevice::ReadOnly | QIODevice::Text))
+    {
+        printErrorMessage(QString("Could not open %1 for reading.").arg(modelFile.fileName()));
+        return false;
+    }
+
+    QStringList modelLines;
+    while (!modelFile.atEnd())
+    {
+        QString line = modelFile.readLine();
+        line.chop(1);
+        for (auto it = replaceMap.begin(); it != replaceMap.end(); ++it) {
+            line.replace(it.key(), it.value());
+        }
+        line.replace(R"(")", R"(\")");
+        modelLines.append(line);
+    }
+    modelLines.last().append("\\n");
+    modelFile.close();
+
```

Comment:
I have a feeling that this code is also reused, but if so we could fix that later (use a common utility function) if possible.

Reply:
No, I will fix it now. The `generateModelFile()` member function is actually duplicated in both FMI and exe generator. I will move it to the base class.